tests/integration/test_integration.py

```diff
@@ -673,6 +795,12 @@ async def run_get_ca_certificate_action(ops_test: OpsTest, timeout: int = 60) ->
     return await ops_test.model.get_action_output(action_uuid=action.entity_id, wait=timeout)
 
 
+async def get_vault_ca_certificate(vault_unit: Unit) -> str:
```

Comment:
Can't we use the juju secrets like we do in documentation?

Reply:
We can, but we would have to have complex logic to check if it's coming from `self-signed-certificates` or vault generated. Also wouldn't work to validate the result of `tls-certificates-access` integration since we'd be checking the source and not the end result. This is way simpler and always gives us the live CA that vault sees.